instructor/common.py

```diff
@@ -0,0 +1,604 @@
+            course = form.save(commit=False)
+            course.owner_id = user.id
+            course.save()
+
+        '''Create a new assignment'''
+        assignment_name = request.session['assignment_name']
+        assignment_id = assignment_name + datetime.datetime.now().strftime("%I:%M%p on %B %d, %Y")
+
+        if request.session['based_on']:
+            a = models.Assignment(course=course,
+                              name=assignment_name,
+                              assignment_id=assignment_id,
+                              basedOn=request.session['based_on'])
+        else:
+            a = models.Assignment(course=course,
+                              name=assignment_name,
+                              assignment_id=assignment_id)
+        a.save()
+        request.session['assignment_id'] = a.id
+        request.session['new'] = False
+
+        if 'save' in request.POST:
+            return redirect("common_course_modify")
+        else:
+            '''Continue'''
+            return redirect('common_assignments_edit_strains')
+
+    else:
+        all_courses = models.Course.objects.all()
+        form = CourseForm()
+        return render_to_response('instructor/course_setup.html',
+                                  {'courses': all_courses,
+                                   'form': form,
+                                   'new': request.session['new']},
+                                  context_instance=RequestContext(request))
+
+
+def assignments_edit(request, assignment_pk):
+    assignment = models.Assignment.objects.get(pk=assignment_pk)
+    if assignment.basedOn is not None:
+        request.session['based_on'] = assignment.basedOn.id
+    else:
+        request.session['based_on'] = ''
+    request.session['assignment_id'] = assignment_pk
+
+    request.session['new'] = False
+
+    return redirect('common_assignment_modify')
+
+
+def assignment_modify(request):
+    assignment_id = request.session['assignment_id']
+    assignment = models.Assignment.objects.get(pk=assignment_id)
+    AssignmentForm = modelform_factory(models.Assignment, fields=['name'])
+
+    if request.method == "POST":
+
+            form = AssignmentForm(request.POST, instance=assignment)
+            form.save()
+
+    all_assignments = models.Assignment.objects.all()
+    if 'continue' in request.POST:
+        return redirect("common_course_modify")
+
+    form = AssignmentForm(instance=assignment)
+    return render_to_response('instructor/assignment_modify.html',
+                              {'assignments': all_assignments,
+                               'form': form,
+                               'based_on': request.session['based_on'],
+                               'new': request.session['new']},
+                              context_instance=RequestContext(request))
+
+def course_modify(request):
+    assignment_id = request.session['assignment_id']
+    assignment = models.Assignment.objects.get(pk=assignment_id)
```

Comment:
I would use `get_object_or_404` here, or do a try..catch and handle the case of missing assignment id gracefully